text/0000-embedded-refract.md

```diff
@@ -0,0 +1,361 @@
+
+```json
+{
+  "first_name": "John",
+  "last_name": "Doe",
+  "age": {
+    "refract": {
+      "element": "number",
+      "meta": {
+        "id": "bowner-103-age"
+      },
+      "content": 28
+    }
+  },
+  "scores": [150, 202, 145]
+}
+```
+
+Above, I've added an ID to the age of the bowler. Notice how I just refracted that value in that specific place. This is also the same as:
+
+```json
+{
+  "refract": {
+    "element": "object",
+    "meta": {
+      "id": "bowler-103"
+    },
+    "content": {
+      "age": {
+        "element": "number",
+        "meta": {
+          "id": "bowner-103-age"
+        },
+        "content": 28
+      }
+    }
+  },
+  "first_name": "John",
+  "last_name": "Doe",
+  "scores": [150, 202, 145]
+}
+```
+
+This will also support full member elements, so even this below is equivalent (note that it's a slight modification from the full Refract):
+
+```json
+{
+  "refract": {
+    "element": "object",
+    "meta": {
+      "id": "bowler-103"
+    },
+    "content": [
+      {
+        "element": "member",
+        "content": {
+          "key": "age",
+          "value": {
+            "refract": {
+              "element": "number",
+              "meta": {
+                "id": "bowner-103-age"
+              },
+              "content": 28
+            }
+          }
+        }
+      }
+    ]
+  },
+  "first_name": "John",
+  "last_name": "Doe",
+  "scores": [150, 202, 145]
+}
+```
+
+In full Refract, you would have to also provide the element for the key "age" in the previous example, but since we can rely on the `refract` keyword, we can leave it unrefracted. We could Refract it more later if necessary.
+
+## Refracting Array and Array Items
+
+Array items can be embedded without refracting the entire array.
+
+```json
+{
+  "first_name": "John",
+  "last_name": "Doe",
+  "age": 28,
+  "scores": [
+    {
+      "refract": {
+        "element": "number",
+        "meta": {
+          "id": "bowler-103-game-1"
+        },
+        "content": 150
+      }
+    },
+    202,
+    145]
+}
+```
+
+Full arrays can also be refracted.
+
+```json
+{
+  "first_name": "John",
+  "last_name": "Doe",
+  "age": 28,
+  "scores": {
+    "refract": {
+      "element": "array",
+      "meta": {
+        "id": "bowler-103-scores"
+      },
+      "content": [150, 202, 145]
+    }
+  }
+}
+```
+
+## Refracting Attributes
+
+This makes it nice for refracting attributes, because now you know what attributes have been refracted. This could become:
+
+```json
+{
+  "refract": {
+    "element": "object",
+    "meta": {
+      "id": "bowler-103"
+    },
+    "attributes": {
+      "foo": "baz"
+    }
+  },
+  "first_name": "John",
+  "last_name": "Doe",
+  "age": 28,
+  "scores": [150, 202, 145]
+}
+```
+
+```json
+{
+  "refract": {
+    "element": "object",
+    "meta": {
+      "id": "bowler-103"
+    },
+    "attributes": {
+      "foo": {
+        "refract": {
```

Comment:
Well, normal Refract can be in normal Refract, so this would be a problem for normal Refract. The issue is there already, but this actually resolves part of the issue of not knowing what is/isn't refracted.

Is your concern the recursion?